go/vt/worker/sharding_key.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package worker
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/youtube/vitess/go/sqltypes"
+
+	"github.com/youtube/vitess/go/vt/key"
+	"github.com/youtube/vitess/go/vt/mysqlctl/tmutils"
+	tabletmanagerdatapb "github.com/youtube/vitess/go/vt/proto/tabletmanagerdata"
+	topodatapb "github.com/youtube/vitess/go/vt/proto/topodata"
+	"github.com/youtube/vitess/go/vt/topo"
+)
+
+// This file defines the interface and implementations of sharding key resolvers.
+
+// shardingKeyResolver defines the interface that needs to be satisifed to get a sharding
+// key from a database row.
+type shardingKeyResolver interface {
+	// shardingKey takes a table row, and returns the sharding key as bytes.
+	// It will return an error if no sharding key can be found.
+	shardingKey(tableName string, row []sqltypes.Value) ([]byte, error)
+}
+
+// v2Resolver is the sharding key resolver that is used by VTGate V2 deployments.
+// In V2, the sharding key column name and type is the same for all tables,
+// and the sharding key value is stored in the database directly.
+type v2Resolver struct {
```

Comment:
if this is per table, you just need to store the index of the column, that's it.